homeassistant/components/tplink/light.py

```diff
@@ -213,13 +219,31 @@ def update(self):
                 except KeyError:
                     # device returned no daily/monthly history
                     pass
+        except (SmartDeviceException, OSError) as ex:
+            _LOGGER.warning(
+                "Retrying in %s for %s|%s due to: %s",
+                SLEEP_TIME,
+                self.smartbulb.host,
+                self._alias,
+                ex,
+            )
+        self._is_ready = True
+
+    async def async_update(self):
+        """Update the TP-Link Bulb's state."""
+        for update_attempt in range(MAX_ATTEMPTS):
+            self._is_ready = False
 
-            self._available = True
+            await self.hass.async_add_executor_job(self.attempt_update)
+            await asyncio.sleep(SLEEP_TIME)
```

Comment:
We should probably move the sleep down after the `_is_ready` check.

Reply:
Makes sense done.